streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java

```diff
@@ -136,15 +141,22 @@ public RocksDBCacheEntry(V value, boolean isDirty) {
         }
     }
 
+    @SuppressWarnings("unchecked")
     public void openDB(ProcessorContext context) {
-        this.context = context;
-        this.dbDir = new File(new File(this.context.stateDir(), parentDir), this.name);
+        // we need to construct the serde while opening DB since
+        // it is also triggered by windowed DB segments without initialization
+        this.serdes = new StateSerdes<>(name,
+                keySerde == null ? (Serde<K>) context.keySerde() : keySerde,
+                valueSerde == null ? (Serde<V>) context.valueSerde() : valueSerde);
+
+        this.dbDir = new File(new File(context.stateDir(), parentDir), this.name);
         this.db = openDB(this.dbDir, this.options, TTL_SECONDS);
     }
 
-    @SuppressWarnings("unchecked")
     public void init(ProcessorContext context, StateStore root) {
-        // first open the DB dir
+        this.context = context;
```

Comment:
This field isn't used anywhere, we can get rid of this line and remove the field from this class.